src/ansi-c/c_nondet_symbol_factory.cpp

```diff
@@ -216,6 +233,25 @@ void symbol_factoryt::gen_nondet_init(
   }
 }
 
+void symbol_factoryt::gen_nondet_array_init(
+  code_blockt &assignments,
+  const exprt &expr,
+  std::size_t depth,
+  const recursion_sett &recursion_set)
+{
+  auto const &array_type = to_array_type(expr.type());
+  auto const array_size = numeric_cast_v<mp_integer>(array_type.size());
+  DATA_INVARIANT(array_size >= 0, "Arrays should have non-negative size");
+  for(auto index = mp_integer(0); index < array_size; ++index)
+  {
+    gen_nondet_init(
+      assignments,
+      index_exprt{expr, from_integer(index, size_type())},
```

Comment:
Nit pick: use parentheses, not braces. Just because we don't use them otherwise.

Reply:
How strongly do you feel about this one? I prefer if we started introducing brace initialisation where we can as it doesn't allow narrowing conversions without a cast.

Reply:
Let's solve this via CODING_STANDARD.md, not guerrilla-style.